util/network-alert/src/notifier.rs

```diff
@@ -60,13 +64,30 @@ impl Notifier {
         if self.noticed_alerts.contains(&alert) {
             return;
         }
-        warn!("receive a new alert: {}", alert.message);
+        self.notify(&alert);
         self.noticed_alerts.push(alert);
         // sort by priority
         self.noticed_alerts
             .sort_by_key(|a| std::u32::MAX - a.priority);
     }
 
+    fn notify(&self, alert: &Alert) {
+        warn!("receive a new alert: {}", alert.message);
+        if let Some(notify_script) = self.config.notify_script.as_ref() {
+            match Command::new(notify_script)
+                .args(&[alert.message.to_owned()])
+                .status()
```

Comment:
The child process may never exit.

Reply:
Not sure this still works https://github.com/alexcrichton/wait-timeout

Reply:
I think we should make it simple, user can set a timeout in his own customized script(IMO it's unnecessary in 99% cases) and decide how to exit child process which better than we kill the script.